### PR TITLE
Feature: BIM Mapping

### PR DESCRIPTION
**Pull request by connorivy**
## Description

- Development toward a solution described in -> https://github.com/specklesystems/speckle-sharp/issues/1248
- This code specifically implements a solution to map types from ETABS/Revit to Revit. There is also code to support tekla, but it looks like we need to change the way we receive tekla objects in order for that code to work. 

## Type of change

- New feature (non-breaking change which adds functionality)

## How has this been tested?

- Manual Tests (please write what did you do?)
I've received several commits using this mapping. However, I have a feeling that there are many corner cases that will not play nice with this feature

## Docs

- Will be updated ASAP
Will update docs before this feature is released

### CODE REVIEW

File: DesktopUI2/DesktopUI2/Views/Windows/Dialogs/MappingViewDialog.xaml [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+      </ScrollViewer>
+      <StackPanel
+        Orientation="Horizontal"
+        HorizontalAlignment="Right">
+        <Button
+          MaxWidth="200"
+          CornerRadius="20"
+          Margin="5"
+          HorizontalAlignment="Left"
+          Command="{Binding ImportFamily}">
```
Comment (teocomi) [inline DesktopUI2/DesktopUI2/Views/Windows/Dialogs/MappingViewDialog.xaml:194]: Very minor remark: I would prefer commands to be named like `ImportFamilyCommand` so it's easier to understand they have a binding in the c# code and avoid the risk of deleting/renaming them involuntarily ;)

File: ConnectorRevit/ConnectorRevit/UI/ConnectorBindingsRevit2/ConnectorBindingsRevit2.Mappings.cs [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+    }
+
+    /// <summary>
+    /// This method creates a mapping dictionary by looping through all of the incoming types and mapping them to the most similar host type of the same category
+    /// </summary>
+    /// <param name="incomingTypesDict"></param>
+    /// <param name="hostTypes"></param>
+    /// <param name="progress"></param>
+    /// <returns></returns>
+    public Dictionary<string, List<MappingValue>> returnFirstPassMap(Dictionary<string, List<string>> incomingTypesDict, Dictionary<string, List<string>> hostTypes, ProgressViewModel progress)
```
Comment (teocomi) [inline ConnectorRevit/ConnectorRevit/UI/ConnectorBindingsRevit2/ConnectorBindingsRevit2.Mappings.cs:89]: Very minor remark #2: function names should be camel case, and they should be private or internal instead of public, unless necessary ofc

**Review by teocomi — APPROVED** [review]
Looks good! I didn't try to understand the mapping logic, so I fully trust you there ;)
Just had 2 minor stylistic comments.

The UI also looks great, I'd just add some margin down here:

image
